engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/content-hosts-bulk-system-purpose-modal.controller.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @ngdoc object
+ * @name  Bastion.content-hosts.controller:ContentHostsBulkSystemPurposeModalController
+ *
+ * @requires $scope
+ * @requires $uibModalInstance
+ * @requires HostBulkAction
+ * @requires Organization
+ * @requires CurrentOrganization
+ * @requires Notification
+ * @requires hostIds
+ *
+ * @description
+ *   A controller for providing bulk action functionality for setting system purpose values
+ */
+angular.module('Bastion.content-hosts').controller('ContentHostsBulkSystemPurposeModalController',
+    ['$scope', '$uibModalInstance', 'HostBulkAction', 'Organization', 'CurrentOrganization', 'Notification', 'hostIds',
+    function ($scope, $uibModalInstance, HostBulkAction, Organization, CurrentOrganization, Notification, hostIds) {
+
+        $scope.organization = Organization.get({id: CurrentOrganization}, function(org) {
+            return org.system_purposes.addons.length;
+        });
+
+        $scope.purposeAddonsList = function () {
+            if ($scope.organization.system_purposes && $scope.organization.system_purposes.addons) {
+                $scope.defaultOptions = ['No Change', 'None (Clear)'];
+                $scope.addons = $scope.organization.system_purposes.addons.map(function (addon) {
+                    return addon; });
+                $scope.addonList = $scope.defaultOptions.concat($scope.addons);
+                return $scope.addonList;
+            }
+            return [];
+        };
+
+        $scope.defaultUsages = ['No change', 'None (Clear)', 'Production', 'Development/Test', 'Disaster Recovery'];
+        $scope.defaultRoles = ['No change', 'None (Clear)', 'Red Hat Enterprise Linux Server', 'Red Hat Enterprise Linux Workstation', 'Red Hat Enterprise Linux Compute Node'];
+        $scope.defaultServiceLevels = ['No change', 'None (Clear)', 'Self-Support', 'Standard', 'Premium'];
+
+        $scope.hostCount = hostIds.included.ids.length;
+
+        $scope.selectedUsages = $scope.defaultUsages[0];
+        $scope.selectedRoles = $scope.defaultRoles[0];
+        $scope.selectedServiceLevels = $scope.defaultServiceLevels[0];
+
+        $scope.selected = {
+           addons: undefined
+        };
+
+        $scope.selectedItemToParam = function (item) {
+            var mapping = {
+                "None (Clear)": "",
+                "No change": null,
+                "": []
+            };
+            if (Array.isArray(item)) {
+                return $scope.selectedAddonsToParam(item);
+            }
+            if (mapping.hasOwnProperty(item)) {
+                return mapping[item];
+            }
+            return item;
+        };
+
+        $scope.selectedAddonsToParam = function (addons) {
+            var selectedIntents = ['No Change', 'None (Clear)'];
```

Comment:
I think `intents` or `intentOptions` is a better name for this variable.  As long as `selected` isn't in the name.

Reply:
Went with `intentOptions `